floater/test/test_generators.py

```diff
@@ -53,6 +53,76 @@ def test_hexmesh(fs):
     # make sure things are "rectangular"
     assert np.allclose(x[0], x[1] + fs.dx/2)
 
+def test_oceancoords(fs):
+    """Verifies that the float grid excludes the masked regions properly."""
+
+    #excludes Anirban's example to speed up the testing time 
+    if fs.dx > 0.02:
+
+        #mask gird is half the spatial resolution of float set
+        mask_lon = np.linspace(fs.xlim[0], fs.xlim[1], int(fs.Nx/2))
+        mask_lat = np.linspace(fs.ylim[0], fs.ylim[1],int(fs.Ny/2))
+        mask = np.ones((np.shape(mask_lon)[0], np.shape(mask_lat)[0]))
```

Comment:
This works, but I feel like the syntax for making the mask is awkward.

Is there a cleaner way to do this?

If we have a pre-defined mask, we probably also have a predefined model grid. In that case, maybe we just want to pass this grid directly to floatset? I know this involves some deeper refactoring of the code, but perhaps that would be for the best.